src/Collection/Fulfilment/OrderNotesCollection.php

```diff
@@ -0,0 +1,93 @@
+<?php
+
+declare(strict_types=1);
+
+namespace MyParcelNL\Sdk\src\Collection\Fulfilment;
+
+use MyParcelNL\Sdk\src\Concerns\HasApiKey;
+use MyParcelNL\Sdk\src\Concerns\HasUserAgent;
+use MyParcelNL\Sdk\src\Model\Fulfilment\OrderNote;
+use MyParcelNL\Sdk\src\Model\MyParcelRequest;
+use MyParcelNL\Sdk\src\Model\RequestBody;
+use MyParcelNL\Sdk\src\Support\Arr;
+use MyParcelNL\Sdk\src\Support\Collection;
+
+class OrderNotesCollection extends Collection
+{
+    use HasUserAgent;
+    use HasApiKey;
+
+    /**
+     * @return array Indexed array of order uuids present in this collection.
+     */
+    private function orderUuidsInCollection(): array
+    {
+        return array_unique($this->reduce(
+            function (array $uuids, OrderNote $orderNote) {
+                $uuids[] = $orderNote->getOrderUuid();
+
+                return $uuids;
+            },
+            []
+        ));
+    }
+
+    /**
+     * @return self Collection of notes that were saved.
+     *
+     * @throws \MyParcelNL\Sdk\src\Exception\AccountNotActiveException
+     * @throws \MyParcelNL\Sdk\src\Exception\ApiException
+     * @throws \MyParcelNL\Sdk\src\Exception\MissingFieldException
+     * @throws \Exception
+     */
+    public function save(): self
+    {
+        $notes = [];
+
+        foreach ($this->orderUuidsInCollection() as $orderUuid) {
+            $newNotes = $this->saveForOrder($orderUuid);
+
+            $notes = array_reduce($newNotes, static function (array $notes, OrderNote $note) {
+                $notes[] = $note;
+
+                return $notes;
+            }, $notes);
+        }
+
+        return (new self($notes));
+    }
+
+    /**
+     * @param  string $orderUuid
+     *
+     * @return array Indexed array holding the saved order notes.
+     * @throws \MyParcelNL\Sdk\src\Exception\AccountNotActiveException
+     * @throws \MyParcelNL\Sdk\src\Exception\ApiException
+     * @throws \MyParcelNL\Sdk\src\Exception\MissingFieldException
+     * @throws \Exception
+     */
+    private function saveForOrder(string $orderUuid): array
+    {
+        $orderNotes = $this->where('orderUuid', '=', $orderUuid)->map(
+            function (OrderNote $orderNote) {
+                $orderNote->validate();
+
+                return $orderNote->toApiObject();
+            }
+        )->toArrayWithoutNull();
+
+        $response = (new MyParcelRequest())
+            ->setUserAgents($this->getUserAgent())
+            ->setRequestParameters(
+                $this->ensureHasApiKey(),
+                new RequestBody('order_notes', array_values($orderNotes))
```

Comment:
ipv array_values kun je `->values()` gebruiken op de collection om de keys te resetten

Reply:
Dat weet ik, dan doe je `->values()->toArray()`. Wat ik bedoel is dat je dit ding beter zo lang mogelijk een collection kan laten